schema-registry/src/main/scala/net.manub.embeddedkafka/schemaregistry/embeddedKafkaWithSchemaRegistry.scala

```diff
@@ -0,0 +1,203 @@
+package net.manub.embeddedkafka.schemaregistry
+
+import java.util.Properties
+
+import io.confluent.kafka.schemaregistry.RestApp
+import io.confluent.kafka.schemaregistry.avro.AvroCompatibilityLevel
+import kafka.server.KafkaServer
+import net.manub.embeddedkafka.EmbeddedKafka.servers
+import net.manub.embeddedkafka.{
+  EmbeddedKafka,
+  EmbeddedKafkaConfig,
+  EmbeddedKafkaSupport,
+  EmbeddedServer
+}
+import org.apache.kafka.clients.consumer.{ConsumerConfig, OffsetResetStrategy}
+import org.apache.kafka.clients.producer.ProducerConfig
+
+trait EmbeddedKafkaWithSchemaRegistry
+    extends EmbeddedKafkaWithSchemaRegistrySupport {
+
+  override def baseProducerConfig(
+      implicit config: EmbeddedKafkaConfigWithSchemaRegistry)
+    : Map[String, Object] = EmbeddedKafkaWithSchemaRegistry.baseProducerConfig
+
+  override def baseConsumerConfig(
+      implicit config: EmbeddedKafkaConfigWithSchemaRegistry)
+    : Map[String, Object] = EmbeddedKafkaWithSchemaRegistry.baseConsumerConfig
+}
+
+sealed trait EmbeddedKafkaWithSchemaRegistrySupport
+    extends EmbeddedKafkaSupport[EmbeddedKafkaConfigWithSchemaRegistry] {
+
+  /**
+    * Starts a ZooKeeper instance, a Kafka broker and a Schema Registry app, then executes the body passed as a parameter.
+    *
+    * @param body   the function to execute
+    * @param config an implicit [[EmbeddedKafkaConfig]]
+    */
+  def withRunningKafka[T](body: => T)(
```

Comment:
I'm not a fan of all this code duplication...

Reply:
as above